postgresql/config/functions.sh

```diff
@@ -67,10 +67,18 @@ bootstrap_replica_via_pg_basebackup() {
 }
 
 ensure_dir_ownership() {
-  echo 'Making sure hab user owns var and data paths'
-  chown -RL hab:hab {{pkg.svc_var_path}}
-  chown -RL hab:hab {{pkg.svc_data_path}}
-  chmod 0700 {{pkg.svc_data_path}}/pgdata
+  if [[ $EUID -eq 0 ]]; then
+    # if EUID is root, so we should chown to pkg_svc_user:pkg_svc_group
+    the_user={{pkg.svc_user}}
+    the_group={{pkg.svc_group}}
+  else
+    # else we can only chown to our primary group
+    the_user=`id -un $EUID`
```

Comment:
What would happen here in the off chance that the $EUID doesn't resolve to a name?  should there be a fallback?

minor nitpick:  let's replace backticks with the newer format: https://github.com/koalaman/shellcheck/wiki/SC2006